flink-clients/src/main/java/org/apache/flink/client/program/rest/UrlPrefixDecorator.java

```diff
@@ -100,4 +105,11 @@ public Class<R> getRequestClass() {
     public M getUnresolvedMessageParameters() {
         return decorated.getUnresolvedMessageParameters();
     }
+
+    @Override
+    public Collection<? extends RestAPIVersion<?>> getSupportedAPIVersions() {
```

Comment:
How about simply doing this:
```
    @Override
    public Collection<? extends RestAPIVersion<?>> getSupportedAPIVersions() {
        return decorated.getSupportedAPIVersions();
    }
```
this class has a clear purpose of only decorating the URL, without dealing with anything delated to the features of the decorated object. That would make the `SQLGatewayUrlPrefixDecorator` redundant - the SQL headers that get decorated will simply pass their `SqlGatewayRestAPIVersion.values()` from `SqlGatewayMessageHeaders` which they already extend.

Reply:
Thanks @afedulov .
Updated the PR. PTAL.